crates/aptos/src/update/prover_dependency_installer.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{BinaryUpdater, UpdateRequiredInfo};
+use crate::update::{
+    prover_dependencies::{REPO_NAME, REPO_OWNER},
+    update_helper::{build_updater, get_path},
+};
+use anyhow::{Context, Result};
+use self_update::update::ReleaseUpdate;
+use std::path::PathBuf;
+
+/// Update Prover dependency.
+#[derive(Debug)]
+pub struct DependencyInstaller {
+    pub binary_name: String,
+
+    pub exe_name: String,
+
+    pub env_var: String,
+
+    pub version_match_string: String,
+
+    pub version_option_string: String,
+
+    pub target_version: String,
+
+    pub install_dir: Option<PathBuf>,
+
+    pub check: bool,
```

Comment:
Some comments on these would be nice, it's not clear what they do.

Reply:
done